src/component_tree/field_extractor.coffee

```diff
@@ -23,13 +23,14 @@ module.exports = class FieldExtractor
 
   onTreeChange: (componentModel) =>
     componentName = componentModel.componentName
-    fieldsToExtract = @metadataConfig.getComponentMap()[componentName]
+    fieldsToExtract = @metadataConfig.getComponentMap()[componentName] || []
+    newlyExtractedFields = @extractFields(fieldsToExtract)
     changedFields = {}
     fieldsHaveChanged = false
 
-    for fieldName, field of @extractFields(fieldsToExtract) when !_.isEqual(@fields[fieldName], field)
+    for fieldName in fieldsToExtract when !_.isEqual(@fields[fieldName], newlyExtractedFields[fieldName])
```

Comment:
I don't like the `when` statement.
It makes the line less readable while it adds too much logic into it.